### DISCUSSION TITLE
What if my time series ('ds') is a little noisy?

### DISCUSSION THREAD

**Discussion by gezabohus** [ourownstory/neural_prophet#1372] [Q&A - get help using NeuralProphet] [answered]
Hi all,

I have one per minute physical observations that I am training on. By the nature of things nothing is perfectly regular, so I have timestamps that vary slightly (say, up to a couple of seconds) from being exactly every 60 seconds.

If I do autoregression, the code fills in missing dates (add_missing_dates_nan in df_utils.py) and NaN as y value. In this case this  means that pretty much everything after the first datapoint becomes NaN.

Is there a way to make a bit less picky or should I preprocess my data to get rid of the noise in ds?

Thanks,

Geza

**Comment by leoniewgnr** [comment, marked as answer]
Yes, you are correct! Its better to preprocess the data, as NP will also be able to handle your data better and do better predictions.

**Reply by gezabohus** [reply]
Thanks!
